Update plantcv recipe to v3.0.4

nfahlgren opened the pull request:
* [x] I have read the [guidelines for bioconda recipes](https://bioconda.github.io/guidelines.html).
* [ ] This PR adds a new recipe.
* [x] AFAIK, this recipe **is directly relevant to the biological sciences** (otherwise, please submit to the more general purpose [conda-forge channel](https://conda-forge.org/docs/)).
* [x] This PR updates an existing recipe.
* [ ] This PR does something else (explain below).

bgruening commented on recipes/plantcv/meta.yaml:
```diff
@@ -2,7 +2,7 @@
 {% set version = "3.0.3" %}
```
should that be 3.0.4?

nfahlgren replied:
Yes, good catch!

nfahlgren replied:
Fixed now, look good otherwise?

bgruening commented:
@nfahlgren what I was wondering is why did if pass before? The URL is changing but the checksum stays?

nfahlgren commented:
@bgruening hmm, not sure. The only thing I changed in the YAML file, in addition to adding the `LICENSE` file, was the `{% set version = "3.0.4" %}` and the `{% set hash_value = "2a120bc5923cd0d273e762bbdc21fef88fa384fc2d5427b671f08d1b71e93b87" %}`. That sha256 checksum is definitely the checksum found for the source distribution at https://pypi.org/project/plantcv/#files and is different than the checksum for 3.0.3 `e8524b622379cf3df03fa1a9a87b7f754b1942329581cc7d94e9d93a3793ca4f`

epruesse commented:
@nfahlgren Can you delete the 3.0.3 folder? I don't think we need that.

nfahlgren commented:
@epruesse for sure, I removed the 3.0.3 folder. I updated the recipe to PlantCV v3.0.5 because we were having issues with the PyPI 3.0.4 version. I didn't mean to push these updates yet though because local tests were giving me trouble, so not sure the circleci tests are going to pass yet.

nfahlgren commented:
There was just a problem with my local environment, tests pass locally again without changing the branch, and looks like it passes on CircleCI too.
